### PR TITLE
Format code with `JuliaFormatter`

### PR DESCRIPTION
**Pull request by mofeing**
This PR formats the code with `JuliaFormatter` and adds a couple of format checking workflows.

### CODE REVIEW

File: .github/workflows/format-check.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: format-check
+
+on:
+  push:
+    branches:
+      - 'master'
+    tags: '*'
+  pull_request:
```
Comment (devmotion) [inline .github/workflows/format-check.yml:8]: IMO the only thing that is needed is a check that is run in every PR. Maybe just copy common setups such as in https://github.com/JuliaDiff/ChainRulesCore.jl/blob/main/.github/workflows/format.yml?

File: .github/workflows/format-pr.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+name: format-pr
```
Comment (devmotion) [inline .github/workflows/format-pr.yml:1]: I would not add such an action, and in particular I don't think it is useful to run it as a cron job. If the check fails in a PR, one can just push a commit with formatting fixes; and if only PRs with correct format are merged, there's no need to check or create PRs on the master branch.

**Review by devmotion — COMMENTED** [review]
Before we format the code base, we have to agree on a code style. IMO we shouldn't rely on general defaults of JuliaFormatter but fix the style that's agreed on in a `.JuliaFormatter.toml` file in the repo.

I tend to use blue style (as e.g. ChainRulesCore: https://github.com/JuliaDiff/ChainRulesCore.jl/blob/main/.JuliaFormatter.toml) but there are of course other common styles.

File: .github/workflows/format-check.yml (hunk 1 above)
Comment (mofeing) [reply .github/workflows/format-check.yml:8]: Oh nice. I didn't know about `reviewdog/action-suggester`. Just refactored to be like the `ChainRulesCore` workflow.

File: .github/workflows/format.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+name: format-check
+
+on:
+  push:
+    branches:
+      - 'master'
+    tags: '*'
```
Comment (devmotion) [inline .github/workflows/format.yml:7]: I think we could skip this and only run the check on PRs:
```suggestion
```

**Review by devmotion — COMMENTED** [review]
Maybe add the blue style badge to the README? Otherwise it looks good to me 🙂

File: .github/workflows/format.yml (hunk 3 above)
Comment (mofeing) [reply .github/workflows/format.yml]: Said and done!

**Pull request closed without merging**